README.md

```diff
@@ -1,64 +1,97 @@
 
-You may of course as well go ahead and download the generated files [here](https://github.com/dangowrt/linksys-e8450-openwrt-installer/releases).
+For utmost safety (but not absolutely necessary), you are recommended to make a complete backup of the device flash __**before**__ running the installer. (see below "Device Flash Backup Procedure")
 
-## Steps
-#### (assuming the device runs stock firmware and is new or just after factory reset)
+You'll need the below to use the script to generate the installer image:
+* OpenWrt ImageBuilder
+* `libfdt-dev`
+* `cmake`
 
-1. Connect any of the LAN ports of the device directly to the Ethernet port of your PC.
-2. Configure the IP address of the PC to be `192.168.1.254`, netmask `255.255.255.0`, no gateway, no DNS.
+If you are not interested in building yourself, the pre-built files are available [here](https://github.com/dangowrt/linksys-e8450-openwrt-installer/releases).
+
+## Installing OpenWrt
+#### Assuming the device is running stock firmware, and is brand new or just after factory reset.
+
+1. Connect any of the LAN ports of the device directly to the Ethernet port of your computer.
+2. Set the IP address of your computer as `192.168.1.254` with netmask `255.255.255.0`, no gateway, no DNS.
 3. Power on the device, wait about a minute for it to be ready.
-4. Open a web browser and navigate to http://192.168.1.1 and wait for the wizard to come up.
-5. Click *exactly* inside the radio button to confirm you have read the terms and conditions, then abort the wizard.
-6. You should then be greeted by the login screen, the stock password is "admin". (This step might only apply to the Linksys, everything else remains the same) 
+4. Open a web browser, navigate to http://192.168.1.1 and wait for the wizard to come up.
+5. Click *exactly* inside the radio button to confirm the terms and conditions, then abort the wizard.
+6. You should then be greeted by the login screen, the stock password is "admin". (This step may be required for Linksys E8450 only)
 7. Navigate to __Administration__ -> __Firmware Upgrade__
-8. Upload `openwrt-mediatek-mt7622-linksys_e8450-ubi-initramfs-recovery-installer.itb` to vendor web interface upgrade page.
-9. Wait for OpenWrt recovery image to come up.
-10. Login and navigate to __System__ -> __Backup / Flash Firmware__
-11. Upload the `openwrt-mediatek-mt7622-linksys_e8450-ubi-sysupgrade.itb` file.
-12. Reboot and proceed to a normal OpenWrt setup (or upload your configuration file).
+8. Upload `openwrt-mediatek-mt7622-linksys_e8450-ubi-initramfs-recovery-installer.itb`.
+9. Wait for a few minutes, the OpenWrt recovery image should come up.
+10. Login and navigate to __System__, save a copy of each of the `mtdblock`.
+11. Upload `openwrt-mediatek-mt7622-linksys_e8450-ubi-squashfs-sysupgrade.itb`.
+12. The device will reboot, you may proceed to setup OpenWrt.
 
-## Upgrading to OpenWrt snapshot release.
+## Upgrading to the Latest OpenWrt Snapshot Release.
```

Comment:
```suggestion
## Upgrading to the latest OpenWrt snapshot.
```
It was already wrong to call a "snapshot" a "release" as it is exactly *not* a release, but only a nightly development snapshot. Using capitals here makes it worse imho...